### PR TITLE
Oracle Collection - V2

### PR DESCRIPTION
**Pull request by mustermeiszer**
# Description

[description]

Fixes #(issue)

## Changes and Descriptions

[List your changes here]

# Checklist:

- [ ] I have added Rust doc comments to structs, enums, traits and functions
- [ ] I have made corresponding changes to the documentation
- [ ] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: pallets/oracle-collection/src/lib.rs [hunk 1]
```diff
@@ -518,26 +693,60 @@ pub mod types {
...
+		Neutral,
+		Dropable,
+	}
+
 	/// Type containing the associated info to a key
 	#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
 	#[scale_info(skip_type_params(T))]
 	pub struct KeyInfo<T: Config> {
-		pub feeders: BoundedBTreeSet<T::FeederId, T::MaxFeedersPerKey>,
+		pub feeders: BoundedBTreeSet<FeederInfo<T>, T::MaxFeedersPerKey>,
```
Comment (lemunozm) [inline pallets/oracle-collection/src/lib.rs:731]: We can have duplicated `FeederId` with different timestamps for different `Relevance`. Are we ok with it?

File: pallets/oracle-collection/src/lib.rs [hunk 2]
```diff
@@ -346,53 +521,42 @@ pub mod pallet {
...
 			Ok(collection)
 		}
 
 		fn register_id(key: &T::OracleKey, collection_id: &T::CollectionId) -> DispatchResult {
-			Keys::<T>::mutate(collection_id, key, |info| {
-				if info.usage_refs.is_zero() {
-					Self::deposit_event(Event::<T>::AddedKey {
-						collection_id: *collection_id,
-						key: *key,
-					});
```
Comment (lemunozm) [inline pallets/oracle-collection/src/lib.rs:393]: Not sure if the event has been removed by error

File: pallets/oracle-collection/src/lib.rs [hunk 3]
```diff
@@ -319,20 +422,92 @@ pub mod pallet {
...
+				},
+			);
+
+			Ok(())
+		}
+
+		/// Proposes a key to be used in the collection
+		#[pallet::weight(T::WeightInfo::set_collection_info())]
+		#[pallet::call_index(6)]
+		pub fn add_key(
```
Comment (lemunozm) [inline pallets/oracle-collection/src/lib.rs:461]: Ideally, is this called by the borrower before `borrow()`?

File: pallets/oracle-collection/src/lib.rs [hunk 4]
```diff
@@ -319,20 +422,92 @@ pub mod pallet {
...
+			Keys::<T>::insert(
+				&collection_id,
+				&key,
+				KeyInfo::new(
+					feeders
+						.try_into()
+						.expect("Size of vec matches size of set at most. qed."),
+					min_feeders.unwrap_or_default(),
+				),
 			);
```
Comment (lemunozm) [inline pallets/oracle-collection/src/lib.rs:504]: Other editor for the same collection can overwrite what other editors has set for the same key

File: pallets/oracle-collection/src/lib.rs [hunk 5]
```diff
@@ -319,20 +422,92 @@ pub mod pallet {
...
+					)
+				)),
+				Error::<T>::IsNotEditor
+			);
+
+			CollectionInfo::<T>::try_mutate(&collection_id, |info| {
+				let info = info.as_mut()?;
+
+				for feeder in &feeders {
+					ensure!(info.feeders.contains(feeder), Error::<T>::FeederNotAllowed);
```
Comment (lemunozm) [inline pallets/oracle-collection/src/lib.rs:490]: I do not understand this condition. Why I can not reuse a feeder?

File: pallets/oracle-collection/src/lib.rs [hunk 6]
```diff
@@ -319,20 +422,92 @@ pub mod pallet {
...
+					Self::maybe_adjust_max_liftime(info, &feeder);
```
Comment (lemunozm) [inline pallets/oracle-collection/src/lib.rs:491]: A borrower could set a `FeederInfo.max_age = 0`, and force the whole collection to be updated each block. Are we ok with this casuistic?

File: pallets/oracle-collection/src/lib.rs (hunk 3 above)
Comment (lemunozm) [reply pallets/oracle-collection/src/lib.rs:461]: Why is `add_key()` an extrinsic instead of being `register_id()`?

**Review by lemunozm — COMMENTED** [review]
Some first questions!

**Comment by mustermeiszer** [comment]
Not be used

**Pull request closed without merging**